pkg/ingester/ingester.go

```diff
@@ -1401,7 +1401,13 @@ func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*corte
 		return &cortexpb.WriteResponse{}, httpgrpc.Errorf(code, wrapWithUser(firstPartialErr, userID).Error())
 	}
 
-	return &cortexpb.WriteResponse{}, nil
+	writeResponse := &cortexpb.WriteResponse{
+		Samples:    int64(succeededSamplesCount),
+		Histograms: int64(succeededSamplesCount), //TODO(Sungjin1212): Should we count histogram?
```

Comment:
How is this implemented in Prometheus. Why we don't count histogram here?

Reply:
I left `TODO` since we are counting histograms just as we count the sample.
But, the Prometheus is counting native histogram https://github.com/prometheus/prometheus/blob/main/storage/remote/write_handler.go#L424.

Reply:
How about starting to count `histogram` when we introduce `PushV2`?

Reply:
I don't understand the concern here. There is nothing prevent us doing it. We should count native histograms

Reply:
yes, I agree with counting native histograms by changing to `Histograms: int64(nativeHistogramCount)`.
My concern is we are counting samples instead of `native histograms`
https://github.com/cortexproject/cortex/blob/master/pkg/ingester/ingester.go#L1269

Reply:
`succeededSamplesCount` is just an internal state/metric we track. There is nothing user facing. We can either change it or introduce new variables to count normal samples and native histogram samples.

Reply:
Can I make PR to introduce `succeededHistogramCount` and `failedHistogramCount` first?

Reply:
I am fine to split but why it needs a separate PR? We can just add a new int64 variable to count succeeded histograms

Reply:
Maybe some changes are needed like we are tracking ingestionRate by calculating `succeededSamplesCount + ingestedMetadata`.
Also, we can introduce new metrics like `cortex_ingester_ingested_native_histograms_total` and `cortex_ingester_ingested_histograms_failures_total`.
WDYT?

Reply:
OK. I am ok to add the new metrics. They are not blocking this PR so can be done either now or after this change.